src/createTokenBridge.integration.test.ts

```diff
@@ -151,6 +154,45 @@ it(`successfully deploys token bridge contracts through token bridge creator`, a
   expect(tokenBridgeContracts.orbitChainContracts.beaconProxyFactory).not.toEqual(zeroAddress);
   expect(tokenBridgeContracts.orbitChainContracts.upgradeExecutor).not.toEqual(zeroAddress);
   expect(tokenBridgeContracts.orbitChainContracts.multicall).not.toEqual(zeroAddress);
+
+  // set weth gateway
+  const setWethGatewayTxRequest = await createTokenBridgePrepareSetWethGatewayTransactionRequest({
```

Comment:
We have a problem here. This method will check whether the Weth gateway is already registered. And in the testnode, that's already been fixed, so it will throw an error saying that the gateway is already registered.

We can separate the check for the gateway from the actual prepareRequest. If we do that, we can pass tokenBridgeContracts to the check function. 

It feels a bit weird though. Because PrepareTransactionRequest wouldn't perform the check then, and the check function would receive an object of tokenBridgeContracts instead of a rollup or inbox address (like all other functions do). And again, we would be changing the behavior only because of the test.

I can't find any other solution though.

What do you think @spsjvc ?

Reply:
Not applicable anymore since we can now override the TokenBridgeCreator used 💪